core/src/main/java/com/datastax/oss/driver/api/core/CoreConsistencyLevel.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.core;
+
+import com.datastax.oss.protocol.internal.ProtocolConstants;
+import com.google.common.collect.ImmutableMap;
+import java.util.Map;
+
+/** A default consistency level supported by the driver out of the box. */
+public enum CoreConsistencyLevel implements ConsistencyLevel {
```

Comment:
Some possible downsides of this approach:

1. You can't really iterate over all possible ConsistencyLevels without checking the concrete enum types.  What about adding a `values()` method to `ConsistencyLevelRegistry` (and the others) so someone could ask what all the possibly supported ones are?

2. Other enum features like `switch` statements only work with the concrete types, so you would have to cast to evaluate in that way, but that doesn't seem like too huge of a deal.

Reply:
For 1 yeah why not. Registries are internal components and we don't have the need at the moment, but it seems reasonable to add `values()`.